web/src/main/java/org/mskcc/oncotree/api/MainTypesApi.java

```diff
@@ -46,49 +49,11 @@
         meta.setCode(200);
         resp.setMeta(meta);
 
+        Map<String, TumorType> tumorTypes = new HashMap<>();
         Version v = (version == null) ? VersionUtil.getDefaultVersion() : VersionUtil.getVersion(version);
-        resp.setData(CacheUtil.getMainTypesByVersion(v));
+        tumorTypes = CacheUtil.getOrResetTumorTypesByVersion(v);
```

Comment:
We are missing a service layer, so some code like this sits here. In general, I think all data logic (aside from simple formatting/mapping) should move (in the future) out of the controller.